### ISSUE TITLE
Better test coverage

### ISSUE THREAD

**Issue by daveyarwood** [daveyarwood/ezzmq#11]
Lately, I've mostly been working my way through the zguide and porting examples over to Clojure with ezzmq. At this point, the tests are a ways behind the kind of ZMQ features I'm demoing in the examples.

We currently have tests for:

- REQ/REP
- PUB/SUB
- PUSH/PULL
- Polling

We could use some additional tests for:
- Different combinations of REQ, REP, ROUTER and DEALER (see e.g. rtreq, rtdealer)
- PUB/SUB proxy
- ROUTER/based proxy between different types of sockets (REQ, DEALER, etc.)
- Identity
- Envelope-based PUB/SUB (see psenvpub/psenvsub)
- Others?

Basically, we should go through the examples and see which examples it makes sense to translate into test cases.
